runtime/lua/vim/lsp/handlers.lua

```diff
@@ -611,6 +615,28 @@ M['window/showDocument'] = function(_, result, ctx, _)
   return { success = success or false }
 end
 
+---@see https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_inlayHint_refresh
+M['workspace/inlayHint/refresh'] = function(err, _, ctx)
+  if err then
+    return vim.NIL
+  end
+
+  for _, bufnr in ipairs(vim.lsp.get_buffers_by_client_id(ctx.client_id)) do
+    local active = false
+    for _, winid in ipairs(api.nvim_list_wins()) do
+      if api.nvim_win_get_buf(winid) == bufnr then
+        active = true
+        break
+      end
+    end
+    if active then
+      require('vim.lsp.inlay_hint').refresh({ bufnr = bufnr })
```

Comment:
Instead of using an `active` variable you could just put this line in its place inside the loop.